perfkitbenchmarker/scripts/object_storage_api_tests.py

```diff
@@ -187,32 +187,57 @@ class SizeDistributionIterator(object):
     dist: dict. The distribution to draw sizes from.
   """
 
-  # IMPLEMENTATION LIMITATION: right now, this object only supports
-  # point distributions, which contain only a single size.
-
   def __init__(self, dist):
-    if len(dist) != 1:
-      raise NotImplementedError(
-          'General distributions are not implemented yet.')
-
-    size, _ = dist.iteritems().next()
-    self.size = size
+    # This class chooses random numbers by using random.random() to
+    # pick a float in [0.0, 100.0), then converting that into a choice
+    # from our distribution. The data structure is two lists, sizes
+    # and cutoffs. If the random number is in the range [cutoffs[i-1],
+    # cutoffs[i]), then the random choice is sizes[i]. Cutoffs has an
+    # implied starting element 0.0, and its last element must be
+    # 100.0.
+    self.sizes = []
+    self.cutoffs = []
+
+    # Sorting the (size, percent change) pairs by size makes for a
+    # nicer interface and also simplifies unit testing.
+    sorted_dist = sorted(dist.iteritems(), key=lambda x: x[0])
+
+    total_percent = 0.0
+    for size, percent in sorted_dist:
+      total_percent += percent
+      self.sizes.append(size)
+      self.cutoffs.append(total_percent)
+
+    # We store percents instead of probabilities because it avoids any
+    # possible edge cases where the percentages would add to 100.0 but
+    # their equivalent probabilities would have truncated
+    # floating-point representations and would not add to 1.0.
+    assert total_percent == 100.0
```

Comment:
@meteorfox That error should be caught by the PKB driver script, which will print a nice error message if the percentages don't add to 100. This is just a sanity check.